Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -597,7 +597,7 @@ bool Parser::match_invalid_escaped_keyword() const
         return false;
     auto token_value = m_state.current_token.value();
     if (token_value == "await"sv)
-        return m_program_type == Program::Type::Module || m_state.in_async_function_context;
+        return m_program_type == Program::Type::Module || m_state.await_expression_is_valid_context;
```

Comment:
`await_expression_is_valid_context` doesn't sound very nice, how about just what the commit message suggests: `await_expression_is_valid`?

Reply:
Yeah I'm still not super happy with the name, ideally I would like `+await` or something as that is how it is used in the spec. Anyway I'm sure someone will figure out a nice name later.